pulpcore/tests/functional/api/using_plugin/test_content_delivery.py

```diff
@@ -102,3 +103,51 @@ def test_remote_artifact_url_update(
     actual_checksum = hashlib.sha256(downloaded_file.body).hexdigest()
     expected_checksum = expected_file_list[0][1]
     assert expected_checksum == actual_checksum
+
+
+@pytest.mark.parallel
+def test_remote_content_changed_with_on_demand(
+    write_3_iso_file_fixture_data_factory,
+    file_repo_with_auto_publish,
+    file_remote_ssl_factory,
+    file_bindings,
+    monitor_task,
+    file_distribution_factory,
+):
+    """When:
+    1. Sync a remote with fileA(digest=123)
+    2. On remote server, change fileA content: fileA(digest=456)
+
+    Then:
+    3. Get fileA from content app will cause a connection-close/incomplete-response.
+    """
+    # Create a remote that points to a file repository with 3 iso files
+    basic_manifest_path = write_3_iso_file_fixture_data_factory("basic")
+    remote = file_remote_ssl_factory(manifest_path=basic_manifest_path, policy="on_demand")
+
+    # Sync from the remote
+    body = RepositorySyncURL(remote=remote.pulp_href)
+    monitor_task(
+        file_bindings.RepositoriesFileApi.sync(file_repo_with_auto_publish.pulp_href, body).task
+    )
+    repo = file_bindings.RepositoriesFileApi.read(file_repo_with_auto_publish.pulp_href)
+
+    # Create a distribution from the publication
+    distribution = file_distribution_factory(repository=repo.pulp_href)
+
+    # Download the manifest from the remote
+    expected_file_list = list(get_files_in_manifest(remote.url))
+
+    # Overwrite the 3 iso files to have different content/digest
+    write_3_iso_file_fixture_data_factory("basic", overwrite=True)
+
+    # Assert response is incomplete
+    get_url = urljoin(distribution.base_url, expected_file_list[0][0])
+    with pytest.raises(ClientPayloadError, match="Response payload is not completed"):
+        download_file(get_url)
+
+    # Assert with curl just to be sure
```

Comment:
This really is the only comment telling me something: "We don't trust the first assert."
```suggestion
    # Assert again with curl just to be sure.
```